tests/property/test_discrete_log.py

```diff
@@ -80,20 +81,50 @@ def test_cached_one_async(self):
         # Act
         loop = asyncio.new_event_loop()
         (plaintext_again, returned_cache) = loop.run_until_complete(
-            discrete_log_async(ciphertext, cache)
+            compute_discrete_log_async(ciphertext, cache)
         )
         loop.close()
 
         # Assert
         self.assertEqual(plaintext, plaintext_again)
         self.assertEqual(len(cache), len(returned_cache))
 
+    @given(integers(0, 1000))
+    def test_precompute_discrete_log(self, exponent: int) -> None:
+        # Arrange
+        minimum_cache_size = exponent + 1
+        element = g_pow_p(exponent)
+
+        # Act
+        cache = precompute_discrete_log_cache(exponent)
+        (calculated_exponent, _returned_cache) = compute_discrete_log(element, cache)
+
+        # Assert
+        self.assertGreaterEqual(len(cache), minimum_cache_size)
+        self.assertEqual(exponent, calculated_exponent)
+
 
 class TestDiscreteLogClass(BaseTestCase):
     """Discrete log tests"""
 
     @given(integers(0, 1000))
-    def test_cached(self, exp: int):
+    def test_precompute(self, exponent: int) -> None:
+        # Arrange
+        # Due to Singleton it could be bigger on previous run.
+        minimum_cache_size = exponent + 1
+        element = g_pow_p(exponent)
+
+        # Act
+        DiscreteLog().set_lazy_evaluation(False)
+        DiscreteLog().precompute_cache(exponent)
+        calculated_exponent = DiscreteLog().discrete_log(element)
+
+        # Assert
+        self.assertGreaterEqual(len(DiscreteLog().get_cache()), minimum_cache_size)
```

Comment:
This is related to the singleton. The test backtracks and does 998 and then repeats one so greater than or equal to is necessary. This is okay though because we only want to ensure the cache is created correctly and covers this case.